lib/bolt/plugin/puppetdb.rb

```diff
@@ -15,9 +16,77 @@ def hooks
         ['lookup_targets']
       end
 
+      def warn_missing_fact(certname, fact)
+        @logger.warn("Could not find fact #{fact} for node #{certname}")
+      end
+
+      def fact_name(raw_fact)
+        fact = raw_fact.split(".")
+        fact.drop(1) if fact[0] == 'facts'
```

Comment:
We should verify that this starts with "facts"(or certname) and error if it isn't.

Reply:
Ah so not even support if there is not 'facts'?